### ISSUE TITLE
Feature: Autocomplete based on string match anywhere in the task label

### ISSUE THREAD

**Issue by natuk** [gtimelog/gtimelog#201]
Current behaviour:
When entering a new task entry in the textfield, autocomplete only picks up matches if the typed string appears at the *beginning* of the string of an existing task label.

Requested behaviour:
When entering a new task entry in the textfield, autocomplete will pick up matches if the typed string appears *anywhere* in a string of an existing task label.

This makes finding the correct task label easier as a user may not recall how the label starts, but they may recall the it contains a word.

**Comment by mgedmin** [comment]
Yeah, I'm not happy with the way completions work currently.  My primary beef is that it returns too many results; the prefix filtering was intended to reduce them to a more manageable list, but isn't working very well.

What I do is turn off GTK completion via dconf-editor and use prefix-completion via PageUp/PageDown, but that is clearly not the right answer.

The completion popup should be limited to maybe 10-20 best matches from what you've already entered, maybe using some kind of fuzzy completion and clever sorting (newer entries upfront, better matches upfront).

The only thing is, I'm not sure how to implement all this, or when I'll have the energy to experiment.
